src/partest/scala/tools/partest/nest/Runner.scala

```diff
@@ -447,7 +449,8 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
   /** Source files for the given test file. */
   def sources(file: File): List[File] = (
     if (file.isDirectory)
-      file.listFiles.toList filter (_.isJavaOrScala)
+      if (testInfo.kind == "presentation") file.listFiles.toList.filter(_.isJavaOrScala)
```

Comment:
We could also not descend into `src/` subdirectories https://github.com/scala/scala/blob/2c462e1c88b5444e5191c3fb2ade28c07e6fa0c3/src/interactive/scala/tools/nsc/interactive/tests/core/TestSettings.scala#L28

Reply:
Or put all java source files into the top directory - javac doesn't seem to mind.